Flutter_Weather/lib/models/weather_extension.dart

```diff
@@ -54,22 +58,26 @@ class DayWeather{
     return DateFormatter.weekDay(initDate.add(Duration(hours: weathers.first.timepoint)));
   }
 
-  WeatherData ? get weatherNow{
-    try {
-      DateTime now = DateTime.now();
-      return weathers.firstWhere((weatherData) =>
-        now.difference(initDate.add(Duration(hours: weathers.first.timepoint))).inMinutes <= 90);
-    } on Exception catch (e) {
-      Logger.logError(
-      className: "Day Weather",
-      methodName: "get WeatherNow",
-      message: e.toString());
-    }
+  WeatherData get weatherNow{
+    DateTime now = DateTime.now().toLocal();
+    WeatherData weatherNow = weathers.firstWhere((weatherData) =>
+      now.difference(initDate.add(Duration(hours: weatherData.timepoint)).toLocal()).inMinutes <= 90);
+    print(now);
+    DateTime chosenDate = initDate.add(Duration(hours: weatherNow.timepoint));
+    print(chosenDate);
+    print(now.difference(chosenDate).inHours);
+    return weatherNow;
   }
 
   int get upperLimitTemp
     => weathers.map((hourData) => hourData.temp2m).toList().reduce(max);
 
   int get lowerLimitTemp
     => weathers.map((hourData) => hourData.temp2m).toList().reduce(min);
+
+  @override
+  String toString() {
+    super.toString();
```

Comment:
The call to the super class is meaningless in this case. Remove super.toString();